std/zig/parse.zig

```diff
@@ -10,13 +10,26 @@ const Error = ast.Error;
 /// Result should be freed with tree.deinit() when there are
 /// no more references to any of the tokens or nodes.
 pub fn parse(allocator: *mem.Allocator, source: []const u8) !*ast.Tree {
-    var tree_arena = std.heap.ArenaAllocator.init(allocator);
-    errdefer tree_arena.deinit();
+    const tree = blk: {
```

Comment:
What's with these seemingly unrelated changes?

These will probably conflict with the stage2 parser rewrite, I'd rather merge that before this one.

Reply:
See the individual commits: this fixes a bug in the parser where it incorrectly uses the allocator (it makes a copy; and then does some operations on the copy; some on the original: this gets internal allocator state confused).
@hryx knows about this (we talked on IRC), and I think he's okay with it getting merged here.

Reply:
@daurnimator It will certain conflict, so I'd rather it wait till the parser is merged. Sorry if I wasn't clear on IRC, I only meant to acknowledge the commit you linked.